src/lib.rs

```diff
@@ -295,6 +295,25 @@ impl Object {
     }
 }
 
+#[cfg(feature = "parallel")]
+macro_rules! writeln_warning {
+    ($dst:expr $(,)?) => {
+        $crate::writeln_warning!($dst, "")
+    };
+    ($dst:expr, $($arg:tt)*) => {
+        ::std::writeln!($dst, "cargo:warning={}", ::std::format_args!($($arg)*))
+    };
```

Comment:
`format_args` expects at least one argument, with the first one being a literal string:

```suggestion
    ($dst:expr, $s:literal) => {
        ::std::writeln!($dst, "cargo:warning={}", s)
    };
    ($dst:expr, $fmt:literal, $($arg:tt)*) => {
        ::std::writeln!($dst, "cargo:warning={}", ::std::format_args!($fmt, $($arg)*))
    };
```

Reply:
I went with this signature to mirror the macros from `std`. While yours is technically more correct, it produces slightly worse error messages in case the first argument isn't a literal string (because format_args! has special handling for that case).

```rust
1  | macro_rules! a {
   | -------------- when calling this macro
...
15 |     a!(VALUE, "world");
   |        ^^^^^ no rules expected this token in macro call
   |
note: while trying to match meta-variable `$fmt:literal`
  --> src/main.rs:2:6
   |
2  |     ($fmt:literal, $($arg:tt)*) => {
```

vs:

```rust
16 |     b!(VALUE, "world");
   |        ^^^^^
   |
help: you might be missing a string literal to format with
   |
16 |     b!("{} {}", VALUE, "world");
   |        ++++++++
```

Reply:
Thanks, in that case I will keep your signature